src/infill/SpaceFillingTreeFill.cpp

```diff
@@ -0,0 +1,274 @@
+{
+    Point3 model_middle = model_aabb.getMiddle();
+    Point3Matrix transformation = LinearAlg2D::rotateAround(Point(model_middle.x, model_middle.y), fill_angle + 45);
+
+    Polygon infill_poly;
+    if (alternate)
+    {
+        std::vector<const SpaceFillingTree::Node*> nodes;
+        generateTreePathAndDepths(nodes);
+        offsetTreePathAlternating(nodes, shift, pocket_size, use_odd_in_junctions, use_odd_out_junctions, infill_poly);
+    }
+    else
+    {
+        Polygon tree_path;
+        generateTreePath(tree_path);
+        offsetTreePath(tree_path, shift, pocket_size, infill_poly);
+    }
+    infill_poly.applyMatrix(transformation); // apply rotation
+
+    if (zig_zaggify)
+    {
+        Polygons infill_pattern;
+        infill_pattern.add(infill_poly);
+        result_polygons = infill_pattern.intersection(outlines);
+    }
+    else
+    {
+        if (infill_poly.size() > 0)
+        {
+            infill_poly.add(infill_poly[0]);
+        }
+        Polygons infill_pattern;
+        infill_pattern.add(infill_poly);
+        Polygons poly_lines = outlines.intersectionPolyLines(infill_pattern);
+        for (PolygonRef poly_line : poly_lines)
+        {
+            for (unsigned int point_idx = 1; point_idx < poly_line.size(); point_idx++)
+            {
+                result_lines.addLine(poly_line[point_idx - 1], poly_line[point_idx]);
+            }
+        }
+    }
+}
+
+SpaceFillingTreeFill::TreeParams SpaceFillingTreeFill::getTreeParams(coord_t line_distance, AABB3D model_aabb)
+{
+    TreeParams ret;
+    AABB aabb(Point(model_aabb.min.x, model_aabb.min.y), Point(model_aabb.max.x, model_aabb.max.y));
+    const Point aabb_size = aabb.max - aabb.min;
+    coord_t minimal_radius = std::max(aabb_size.X, aabb_size.Y) / 2;
+    minimal_radius *= sqrt(2.0); // to account for any possible infill angle
+
+    /*
+     * For the normal cross infill the width of the cross is equal to the width of the crosses which are left out.
+     * 
+     *  ▉  ▄    white cross left out          r/2
+     * ▀▉▀▀▉▀ ↙       by the black          ^^^^^^^^
+     * ▄▉▄   ▄▉▄                     +--------------+
+     *  ▉  ▄  ▉                      :              |
+     * ▀▉▀▀▉▀▀▉▀▀                    :  -------+    |   quarter of the fractal with depth one
+     *                               :,'       |    | the fractal with depth one
+     *                               :    .'   |    |
+     *                            .  :  .' :   |    |
+     *                          .    :.'...:.,'.....+
+     *                        .            :
+     *                         '-.       . :   :
+     *                          l  '-, .   vvvvv
+     *                                       n
+     * minimum offset = 0
+     * maximum offset is such that n = r /2
+     * 2n^2 = l/2  ==>  n = 1/4 sqrt(2) l
+     * r = .5 sqrt(2) l
+     */
```

Comment:
> I don't see what the "white cross left out by the black" has to do with it.

The opening sentence of the documentation mentions both, so they both have to do with it: "For the normal cross infill the width of the cross is equal to the width of the crosses which are left out."

>The other drawing can probably be improved a bit (improved ASCII art) for clarity. Took me a while to understand its relevance because I didn't see the relation to l.

I don't know how to improve. What do you mean? `l` is in the drawing.

Reply:
I tried to make improvements to the drawing. It's still not very clear, but I tried to make the outline of the line more clear. And I used capital `L` rather than lowercase `l` because it stands out a bit more among the ASCII art, at least in the font I use :)